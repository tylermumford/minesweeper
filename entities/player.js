class Player {
    get name() {return this.#name}
    
    constructor(name) {
        if (name) this.#name = name;
    }

    #name = 'Anonymous'

    toJSON() {
        return {name: this.#name}
    }
}

export default Player;
