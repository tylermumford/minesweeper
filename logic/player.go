package logic

type Player struct {
	PlayerId string
	Name     string
}
