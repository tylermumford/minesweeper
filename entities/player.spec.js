import { createNewPlayer } from "./player.js";

test('should be created', () => {
    const p = createNewPlayer();

    expect(p).toBeDefined();
})

test('should be created with a name', () => {
    const p = createNewPlayer('Quill');

    expect(p.name).toBe('Quill');
})
