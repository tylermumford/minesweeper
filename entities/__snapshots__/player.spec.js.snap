// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player class should serialize to JSON 1`] = `"{\\"name\\":\\"Anonymous\\"}"`;
